app/api/route/message_api.rb

```diff
@@ -0,0 +1,31 @@
+module Route
+  class MessageApi < Base
+    paginate paginate_settings
+
+    params do
+      optional :thread, type: Integer, desc: 'ID of thread'
+      optional :order_by, type: Symbol, values:%i( created_at  ), desc: 'Order of returned issues.'
+      optional :order, type: Symbol, values:%i(asc desc), default: :asc
+      optional :submit_external, type: String, desc: 'Filter by external service submission mark'
+    end
+
+    helpers do
+      def post_or_get_messages
+        scope = Message.joins(:thread).includes(:created_by)
+        scope = scope.where('message_threads.privacy' => 'public')
+        scope = scope.where(censored_at: nil)
+        scope = scope.order(params[:order_by] => params[:order]) if params[:order_by]
+        scope = scope.where(thread_id: params[:thread]) if params[:thread]
+        scope = scope.where('message_threads.submit_external' => params[:submit_external]) if params[:submit_external]
```

Comment:
There is no `submit_external` in the codebase.  Is it something you've added elsewhere?

Reply:
Yes, I am working on function, where threads could be marked to be send to the municipality.
I will remove this from this PR, it belongs to separate PR with that functionality.